caddytest/integration/stream_test.go

```diff
@@ -0,0 +1,201 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/caddyserver/caddy/v2/caddytest"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+)
+
+func TestH2ToH2CStream(t *testing.T) {
```

Comment:
If you could add a comment with a link to https://github.com/caddyserver/caddy/issues/3556, I think that would be helpful if this is revisited in the future!